src/rect.rs

```diff
@@ -420,6 +420,47 @@ impl Rect {
     pub fn to_rounded_rect(self, radius: f64) -> RoundedRect {
         RoundedRect::from_rect(self, radius)
     }
+
+    /// Returns the largest possible `Rect` that is fully contained in this rect with the fixed
+    /// aspect ratio given by `aspect_ratio`.
+    ///
+    /// The resulting rectangle will be centered if it is smaller than the input rectangle.
+    ///
+    /// For the special case where the aspect ratio is `1.0`, the resulting `Rect` will be square.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use kurbo::Rect;
+    /// let outer = Rect::new(0.0, 0.0, 10.0, 20.0);
+    /// let inner = outer.fix_aspect_ratio(1.0);
+    /// // The new `Rect` is a square centered at the center of `outer`.
+    /// assert_eq!(inner, Rect::new(0.0, 5.0, 10.0, 15.0));
+    /// ```
+    ///
+    pub fn fix_aspect_ratio(&self, aspect_ratio: f64) -> Rect {
```

Comment:
The only real Q for me about this PR is about naming.

`constrain_aspect_ratio`? `contained_rect_with_aspect_ratio`? `with_fixed_aspect_ratio`?

Reply:
i think probably a longer name that fully describes the function would be better - so `contained_rect_with_aspect_ratio`, what do others think?

Reply:
I agree that clarity is the thing to optimize for here, that name is meh but also it's fine, happy to get this merged with that change (and optionally something along the lines of the doc suggestion above) and then we can get it out and into the next piet & druid releases.

Reply:
It can always be changed (or removed) before 1.0.